samcli/lib/config/samconfig.py

```diff
@@ -247,6 +248,24 @@ def _deduplicate_global_parameters(self, cmd_name_key, section, key, env=DEFAULT
             # Only keep the global parameter
             del self.document[env][cmd_name_key][section][key]
 
+    def _get_default_file(self, config_dir: str) -> str:
+        """Return a defaultly-named config file, if it exists, otherwise the current default."""
+        config_files_found = 0
+        config_file = DEFAULT_CONFIG_FILE_NAME
+
+        for extension in reversed(self.FILE_MANAGER_MAPPER.keys()):
+            filename = DEFAULT_CONFIG_FILE + extension
+            if Path(config_dir, filename).exists():
+                config_files_found += 1
+                config_file = filename
+
+        if config_files_found == 0:  # Config file doesn't exist (yet!)
+            LOG.debug(f"No config file exists yet. Creating one as {config_file}.")
+        elif config_files_found > 1:  # Multiple config files; let user know which is used
+            LOG.debug(f"More than one samconfig file found. Using {config_file}.")
```

Comment:
Yup, thats really the only concern I had, was whether or not this flag is set for the commands that use a config file, otherwise the flag usage part will get printed, and customers won't actually be able to use that flag.